mount/mount_windows.go

```diff
@@ -9,15 +16,73 @@ var (
 
 // Mount to the provided target
 func (m *Mount) Mount(target string) error {
-	return ErrNotImplementOnWindows
+	home, layerID := filepath.Split(m.Source)
+
+	parentLayerPaths, err := m.GetParentPaths()
+	if err != nil {
+		return err
+	}
+
+	var di = hcsshim.DriverInfo{
+		HomeDir: home,
+	}
+
+	if err = hcsshim.ActivateLayer(di, layerID); err != nil {
+		return errors.Wrapf(err, "failed to activate layer %s", m.Source)
+	}
+	defer func() {
+		if err != nil {
+			hcsshim.DeactivateLayer(di, layerID)
+		}
+	}()
+
+	if err = hcsshim.PrepareLayer(di, layerID, parentLayerPaths); err != nil {
+		return errors.Wrapf(err, "failed to prepare layer %s", m.Source)
+	}
+	defer func() {
+		if err != nil {
+			hcsshim.UnprepareLayer(di, layerID)
+		}
+	}()
+	return nil
+}
+
+// ParentLayerPathsFlag is the options flag used to represent the JSON encoded
+// list of parent layers required to use the layer
+const ParentLayerPathsFlag = "parentLayerPaths="
+
+// GetParentPaths of the mount
+func (m *Mount) GetParentPaths() ([]string, error) {
+	var parentLayerPaths []string
+	for _, option := range m.Options {
+		if strings.HasPrefix(option, ParentLayerPathsFlag) {
+			err := json.Unmarshal([]byte(option[len(ParentLayerPathsFlag):]), &parentLayerPaths)
```

Comment:
Isn't using JSON here still problematic since the comma won't be escaped? What is the main motivation behind encoding with JSON

Reply:
The intent was just to pack the parent layer paths in a single mount in a way that would be able to get back the same list when mounting. JSON was used because it would still be readable, but any way to Marshal should be ok, as long as all characters in the path are OK in the marshaled form.

I don't think comma is a problem, as on Windows this is not actually used by a `mount` syscall similar to Unix, but rather is parsed by containerd and passed as extra info to HCS to assemble the layer. I might be missing somewhere where this struct is serialized though.

Reply:
Fair enough, the serialization from go array should only go to protobuf types and back. We have a `ctr` command which gives the mount command, but we can just have it output something else on Windows since a Unix mount command doesn't make sense here.